Dockerfile

```diff
@@ -1,11 +1,42 @@
-FROM busybox:ubuntu-14.04
-MAINTAINER Jimmi Dyson <jimmidyson@gmail.com>
+FROM golang:1.9.2-alpine3.6
 
-ADD ./stage/tyk /bin/tyk
-ADD tyk.conf.example /etc/tyk/tyk.conf
-ADD templates/ /etc/tyk/templates/
+RUN apk add --update git vim bash && rm -rf /var/cache/apk/*
 
-EXPOSE 8080
+RUN echo 'alias ll="ls -la"' >> ~/.bashrc
 
-ENTRYPOINT ["/bin/tyk"]
-CMD []
+ARG BUNDLE_GITHUB__COM
+RUN echo -e "machine github.com\n  login ${BUNDLE_GITHUB__COM}" >> ~/.netrc
+RUN echo -e '' > ~/.netrc
+
+ARG BUNDLE_GITHUB__COM
+ARG APP_NAME
+ENV APP_NAME ${APP_NAME:-tyk}
+
+ENV PROTOVERSION 3.2.0
+ENV TYKVERSION 2.3.10
+ENV TYKLANG ""
+
+ENV TYKLISTENPORT 8080
+ENV TYKSECRET 352d20ee67be67f6340b4c0605b044b7
+ENV TYK_PATH /opt/tyk-gateway/
+
+LABEL Description="Tyk Gateway docker image" Vendor="Tyk" Version=$TYKVERSION
+
+RUN mkdir -p /go/src/github.com/gtforge/${APP_NAME}
+RUN echo -e "machine github.com\n  login ${BUNDLE_GITHUB__COM}" >> ~/.netrc
+
+RUN echo -e '' > ~/.netrc
+
+ADD . /go/src/github.com/gtforge/${APP_NAME}
+WORKDIR /go/src/github.com/gtforge/${APP_NAME}
+RUN go get && go build -v -i
```

Comment:
please delete the source files from the container after the "go build" command. 
They are not needed at runtime.